torch/csrc/jit/runtime/interpreter.cpp

```diff
@@ -634,11 +715,35 @@ struct CodeImpl {
     }
   }
 
+  void emitLoadInputs(at::ArrayRef<Value*> inputs, int num_include) {
+    int count = 0;
+    for (Value* input : inputs) {
+      if (count < num_include) {
+        emitUse(input, false);
+        count++;
+      } else {
+        emitUse(input, true);
+      }
+    }
+  }
+
   void emitOperator(Node* node) {
-    emitLoadInputs(node->inputs());
     const Operator& op = node->getOperator();
+    auto unique_op_name =
+        op.schema().name() + "." + op.schema().overload_name();
+    auto min_ignore = 0;
+    // make sure we only do this for mobile code
+    if (from_mobile_ &&
+        op_to_unnecessary_args_.find(unique_op_name) !=
+            op_to_unnecessary_args_.end()) {
+      min_ignore = op_to_unnecessary_args_[unique_op_name];
+    }
+
+    auto necessary_size = node->inputs().size() - min_ignore;
+    emitLoadInputs(node->inputs(), necessary_size);
+
     if (op.hasOperation() && op.schema().is_vararg()) {
-      insertInstruction(OPN, operator_table_.size(), node->inputs().size());
+      insertInstruction(OPN, operator_table_.size(), necessary_size);
```

Comment:
@gmagogsfm , is it safe to update vararg operators (OPN)?

Reply:
For vararg operators, our algorithm would return 0 because user would always give inputs. (e.g there won't be any unnecessary arg in the end). Therefore I think the necessary_size will always be same node->inputs.size(). Maybe I shouldn't even change this?

Reply:
Yeah, I agree with you Tugsu that this probably shouldn't be changed.